app/services/taxons_to_legacy_associations_tagging.rb

```diff
@@ -54,8 +54,20 @@ def legacy_mapping_for_taxons(selected_taxons)
   end
 
   def any_legacy_associations_for_edition?(edition)
-    edition.policy_content_ids.any? ||
-      edition.topic_ids.any? ||
-      edition.specialist_sectors.any?
+    tagged_to_policies?(edition) ||
+      tagged_to_policy_areas?(edition) ||
+      tagged_to_specialist_sectors?(edition)
+  end
+
+  def tagged_to_policies?(edition)
+    defined?(edition.policy_content_ids) && edition.policy_content_ids.any?
+  end
+
+  def tagged_to_policy_areas?(edition)
+    defined?(edition.topic_ids) && edition.topic_ids.any?
+  end
+
+  def tagged_to_specialist_sectors?(edition)
+    defined?(edition.specialist_sectors) && edition.specialist_sectors.any?
```

Comment:
Instead of splitting this out in to 3 methods, you could wrap the `any?` call in `.try(...)`, e.g. `edition.try(:specialist_sectors).try(:any?)`. I can't quite decide if it's better, but I thought I'd mention it.

Reply:
I think if there where's only one `try` I'd agree with you, but think that for readability I'd still split it into three methods.

Reply:
Sure :)